### PR TITLE
Define input parameters for image run

### LINKED ISSUE #233: As Alice, I can provide or edit the input parameters when running an image, so that I don't need to create a deployment to achieve that.
### Description

Currently, the only way to provide or modify the input parameters of an image is when it is used as a node within a deployment. For images that make sense without the need of a deployment (e.g. a WordPress machine) it should be possible to provide or modify the input parameters directly from the "Run Image" dialog, in a similar way that is done for nodes in the "Run Deployment dialog".

### Upstream dependency

- [x] SlipStream/SlipStreamServer#170: Allow users to redefine "input parameters" for "Simple Run"

### PR DESCRIPTION
**Pull request by rbf**
Related to #233.

### CODE REVIEW

File: clj/resources/lang/en.edn [hunk 1]
```diff
@@ -510,6 +510,9 @@
    :slipstream.ui.models.parameters.launch-mutable-run?.description             :global.nouns.mutable-deployment
    :slipstream.ui.models.parameters.launch-mutable-run?.help-hint               "Enabling this option will launch a _mutable_ deployment, which will allow the `multiplicity` for each node to be changed after launch. **Warning!** This is a _BETA_ feature!"
 
+   :slipstream.ui.models.parameters.image-target-cloud.description              :global.nouns.cloud
+   :slipstream.ui.models.parameters.image-target-cloud.help-hint                "Choose the cloud to deploy the image to. The cloud flagged with a `*` is the default configured in your user profile."
```
Comment (schaubl) [inline clj/resources/lang/en.edn]: `Choose the cloud where to deploy the image.` ?

Comment (rbf) [inline clj/resources/lang/en.edn]: > `Choose the cloud where to deploy the image.` ?

@loomis ?

Comment (loomis) [inline clj/resources/lang/en.edn]: "Choose the cloud where the image will be deployed."

File: clj/resources/lang/en.edn [hunk 2]
```diff
@@ -510,6 +510,9 @@
...
+
    :slipstream.ui.models.parameters.deployment-target-cloud.description         :global.nouns.cloud
    :slipstream.ui.models.parameters.deployment-target-cloud.help-hint           "Choose the cloud to deploy to. The cloud flagged with a `*` is the default configured in your user profile. The option `Specify for each node` will reveal this menu for each of the nodes below."
```
Comment (rbf) [inline clj/resources/lang/en.edn]: @loomis 
Thanks for the correction above.
What about the equivalent string for the deployments: `Choose the cloud to deploy to.` ?
=> `Choose the cloud where the deployment will be deployed.` ? :unamused:

**Comment by schaubl** [comment]
It doesn't work for me.
When I click to the `Run image` button, nothing happens and no request is sent.

On the JS console I got this error when the page is loaded:
```
Error: Syntax error, unrecognized expression: input:hidden[name=parameter--webserver.hostname] 
jquery-1.11.1.min.js:2:12719
```
May be related to this line: https://github.com/slipstream/SlipStreamUI/blob/issue_%23233-redefine_input_parameters/clj/resources/static_content/js/util.js#L801

File: clj/resources/lang/en.edn (hunk 2 above)
Comment (loomis) [inline clj/resources/lang/en.edn]: I'd try "Choose the cloud to use for the deployment."  to avoid repetition.

**Comment by rbf** [comment]
@schaubl 
Looking into it. Cheers.

File: clj/resources/lang/en.edn (hunk 2 above)
Comment (rbf) [inline clj/resources/lang/en.edn]: Changed in  bd93474.
Thanks.

File: clj/resources/lang/en.edn (hunk 1 above)
Comment (rbf) [inline clj/resources/lang/en.edn]: Changed in  bd93474.
Thanks.

**Comment by rbf** [comment]
@schaubl 
I pushed a fix for the issue regarding your [last comment](https://github.com/slipstream/SlipStreamUI/pull/276#issuecomment-91055551). Thanks for pinpointing the exact LOC with the issue.

Please let me know if that fixes your issue and the PR can be merged.

I still have 2 questions though:

1. How did you encounter this issue? The `webserver.hostname` looks like an output parameter that should not be in the form. Can you also share a screenshot of the issue please?

1. To check this I created a parameter with several word-separators (dot, dash, double-dash and underscore) and discovered that the server fails to correctly parse the form if there is a double-dash `--` in the parameter name. I created #277 for this.

Cheers.

**Comment by schaubl** [comment]
@rbf 
1. I tried to run the [apache](https://slipstream.sixsq.com/module/examples/tutorials/service-testing-windows/apache) image of the [service-testing-windows](https://slipstream.sixsq.com/module/examples/tutorials/service-testing-windows). I cannot share a screenshot because I got no error message. Clicking on the `Run image` button just do nothing.
2. Ok

Cheers

**Comment by schaubl** [comment]
Ok, I tested again and now it's working.

**Pull request merged**